ext/json/json_parser.tab.c

```diff
@@ -1882,12 +1997,17 @@ int php_json_parser_object_update(php_json_parser *parser, zval *object, zend_st
 		ZVAL_NEW_STR(&zkey, key);
 		zend_std_write_property(object, &zkey, zvalue, NULL); 
 
-		if (Z_REFCOUNTED_P(zvalue)) {
-			Z_DELREF_P(zvalue);
-		}
+		zval_dtor(zvalue);
 	}
 	zend_string_release(key);
 
+	if (EG(exception)) {
+		parser->scanner.errcode = PHP_JSON_ERROR_DEPTH;
```

Comment:
In general, adding a new JSON error isn't worth it, as, like Nikita says:

> Getting C=1000
collisions for a single hash-value is extremely unlikely until you
explicitly design your keys to trigger this situation.

Thus it's not worth it.